src/main.rs

```diff
@@ -37,68 +41,135 @@ fn show_version() {
 
 pub fn main() {
     // Check for version and help flags even when invoked as 'cargo-clippy'
-    if std::env::args().any(|a| a == "--help" || a == "-h") {
+    if env::args().any(|a| a == "--help" || a == "-h") {
         show_help();
         return;
     }
 
-    if std::env::args().any(|a| a == "--version" || a == "-V") {
+    if env::args().any(|a| a == "--version" || a == "-V") {
         show_version();
         return;
     }
 
-    if let Err(code) = process(std::env::args().skip(2)) {
-        std::process::exit(code);
+    if let Err(code) = process(env::args().skip(2)) {
+        process::exit(code);
     }
 }
 
-fn process<I>(mut old_args: I) -> Result<(), i32>
-where
-    I: Iterator<Item = String>,
+struct ClippyCmd {
+    unstable_options: bool,
+    cmd: &'static str,
+    args: Vec<String>,
+    clippy_args: String
+}
+
+impl ClippyCmd
 {
-    let mut args = vec!["check".to_owned()];
+    fn new<I>(mut old_args: I) -> Self
+    where
+        I: Iterator<Item = String>,
+    {
+        let mut cmd = "check";
```

Comment:
Maybe `cargo_subcommand` is more descriptive here?
```suggestion
        let mut cargo_subcommand = "check";
```